source/server-selection/tests/logging/replica-set.yml

```diff
@@ -0,0 +1,223 @@
+                type: RSPrimary
+          count: 1
+      - name: insertOne
+        object: *collection
+        arguments:
+          document: { x : 1 }
+    expectLogMessages:
+      - client: *client
+        messages:
+          - level: debug
+            component: serverSelection
+            data:
+              message: "Server selection started"
+              selector: { $$exists: true }
+              operation: insert
+              topologyDescription: { $$exists: true } 
+          - level: debug
+            component: serverSelection
+            data:
+              message: "Server selection succeeded"
+              selector: { $$exists: true }
+              operation: insert
+              topologyDescription: { $$exists: true }
+              serverHost: { $$type: string }
+              serverPort: { $$type: [int, long] }
+
+  - description: "Server selection fails due to unsatisfiable read preference"
+    runOnRequirements:
+      - minServerVersion: "4.0" # min version for failCommand on mongod
+    operations:
+      # ensure we've discovered the primary before starting, so we don't hav extra logs
+      # from the driver performing server selection to determine session support status.
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          client: *client
+          event:
+            serverDescriptionChangedEvent:
+              newDescription:
+                type: RSPrimary
+          count: 1
+      # the `find` should never succeed due to the unsatisfiable read preference.
+      - name: find
+        object: *unsatisfiableRPColl
+        arguments:
+          filter: { x : 1 }
+        expectError:
+          isClientError: true # server selection error
+    expectLogMessages:
+      # the topology won't change during this time, so although we exhaust
+      # serverSelectionTimeoutMS we only expect a single message.
+      - client: *client
+        messages:
+          - level: debug
+            component: serverSelection
+            data:
+              message: "Server selection started"
+              selector: { $$exists: true }
+              operation: find
+              topologyDescription: { $$exists: true } 
+          - level: info
+            component: serverSelection
+            data:
+              message: "Waiting for suitable server to become available"
+              selector: { $$exists: true }
+              operation: find
+              topologyDescription: { $$exists: true } 
+              remainingTimeMS: { $$type: [int, long] }
+          - level: debug
+            component: serverSelection
+            data:
+              message: "Server selection failed"
+              selector: { $$exists: true }
+              operation: find
+              topologyDescription: { $$exists: true }
+              failure: { $$exists: true }
+
+  - description: "Waiting message is logged when a suitable server is not immediately available"
```

Comment:
This test is racy (server can be rediscovered before selection), so I think having "Failure due to unreachable server" instead is sufficient. cc @kmahar.

Reply:
Sorry, looks like removing sessions check helped with the timings. Ran the tests for a while, not issues so far.

Reply:
It's still racy after all, [example](https://evergreen.mongodb.com/test_log/dot_net_driver_unsecure_tests__version~5.0_os~windows_64_topology~sharded_cluster_auth~noauth_ssl~nossl_test_net472_patch_1f761191fb44d0c53dcbfe5d739a20f057be6c7e_6466c28ca4cf472554c992b6_23_05_19_00_28_26/0?test_name=05b3b956d9eec9c7ad72aa7f863e74f5&group_id=#L1).
Although it's quite rare 2% of cases, and only on [EG](https://spruce.mongodb.com/version/6466c28ca4cf472554c992b6/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC). 
I think` Failure due to unreachable server` is sufficient to test this code path.

Reply:
ah I see. I tried to do the high heartbeat frequency to prevent that but maybe it still isn't enough on occasion. I agree if we are covering the waiting message via the other test then it seems ok to remove this one.